packages/ic-management/src/ic-management.canister.ts

```diff
@@ -221,4 +226,40 @@ export class ICManagementCanister {
 
     return canister_id;
   };
+
+  /**
+   * Given a `get_utxos_request`, which must specify a Bitcoin address and a Bitcoin network (`mainnet` or `testnet`), the function returns all unspent transaction outputs (UTXOs) associated with the provided address in the specified Bitcoin network based on the current view of the Bitcoin blockchain available to the Bitcoin component.
+   *
+   * @link https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-bitcoin_get_utxos
+   *
+   * @param {Object} params
+   * @param {BitcoinNetwork} params.network Tesnet or mainnet.
+   * @param {Object} params.filter The optional filter parameter can be used to restrict the set of returned UTXOs, either providing a minimum number of confirmations or a page reference when pagination is used for addresses with many UTXOs.
+   * @param {string} params.address A Bitcoin address.
+   * @returns {Promise<bitcoin_get_utxos_result>} The UTXOs are returned sorted by block height in descending order.
+   */
+  bitcoinGetUtxos = (
+    params: BitcoinGetUtxosParams,
+  ): Promise<bitcoin_get_utxos_result> => {
+    const { bitcoin_get_utxos } = this.service;
+    return bitcoin_get_utxos(toBitcoinGetUtxosParams(params));
+  };
+
+  /**
+   * This method is identical to `bitcoinGetUtxos`, but exposed as a query.
+   *
+   * @link https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-bitcoin_get_utxos_query
+   *
+   * @param {Object} params
+   * @param {BitcoinNetwork} params.network Tesnet or mainnet.
+   * @param {Object} params.filter The optional filter parameter can be used to restrict the set of returned UTXOs, either providing a minimum number of confirmations or a page reference when pagination is used for addresses with many UTXOs.
+   * @param {string} params.address A Bitcoin address.
+   * @returns {Promise<bitcoin_get_utxos_result>} The UTXOs are returned sorted by block height in descending order.
+   */
+  bitcoinGetUtxosQuery = (
```

Comment:
Did you think of implementing one method with `certified` parameter and call different endpoints inside it instead of exposing both endpoints directly?

Reply:
I thought about it too but, I discarded it for following two reasons:

- the parameters and returns of `bitcoinGetUtxos` and `bitcoinGetUtxosQuery` are different. while they are similar in term of content, the did file specify two different implements / different types. so while we might be able to merge it now, we have no guarantee that this will be always the case.
- the `IcMgmtCanister` does not have yet a certified service yet
- the `IcMgmtCanister` is not really a canister, it's kind of particular

For above reasons, felt like there were more valid reason to not used a `certified` parameter and have distinctive functions but, I'm definitely open about it if you rather like this approach.

Reply:
Thanks for the explanation! That makes sense to me.